apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnForeignKey.tsx

```diff
@@ -116,13 +116,20 @@ const ColumnForeignKey = ({
           table={{
             id: table.id,
             name: table.name,
-            columns:
-              column.isNewColumn && column.name
-                ? (table.columns as any[]).concat(column)
-                : (table.columns as any[]).map((c) => {
-                    if (c.id === column.id) return { ...c, name: column.name }
-                    else return c
-                  }),
+            columns: (column.isNewColumn && column.name
+              ? (table.columns ?? []).concat(column as any)
```

Comment:
`any` ?

Reply:
it was `any` before so not a blocker.

Reply:
hmm actually lemme just clean this up too

Reply:
okay pushed a refactor up!